operator/pkg/cli/migratev2/manager.go

```diff
@@ -0,0 +1,188 @@
+
+	logf("Deleting manager install jobs")
+	err = deleteManagerInstallJobs(ctx, cli, nodes.Items)
+	if err != nil {
+		return fmt.Errorf("delete jobs: %w", err)
+	}
+	logf("Successfully deleted manager install jobs")
+
+	return nil
+}
+
+func getOperatorImageName(ctx context.Context, cli client.Client, in *ecv1beta1.Installation) (string, error) {
+	if in.Spec.AirGap {
+		err := metadata.CopyVersionMetadataToCluster(ctx, cli, in)
+		if err != nil {
+			return "", fmt.Errorf("copy version metadata to cluster: %w", err)
+		}
+	}
+
+	meta, err := release.MetadataFor(ctx, in, cli)
+	if err != nil {
+		return "", fmt.Errorf("get release metadata: %w", err)
+	}
+
+	for _, image := range meta.Images {
+		if strings.Contains(image, "embedded-cluster-operator-image") {
+			return image, nil
+		}
+	}
+	return "", fmt.Errorf("no embedded-cluster-operator image found in release metadata")
+}
+
+func ensureManagerInstallJobForNode(
+	ctx context.Context, cli client.Client,
+	node corev1.Node, in *ecv1beta1.Installation, operatorImage string,
+	licenseSecret string, appVersionLabel string,
+) (string, error) {
+	existing, err := getManagerInstallJobForNode(ctx, cli, node)
+	if err == nil {
+		if managerInstallJobHasSucceeded(existing) {
+			return existing.Name, nil
+		} else if managerInstallJobHasFailed(existing) {
+			err := cli.Delete(ctx, &existing)
+			if err != nil {
+				return "", fmt.Errorf("delete job %s: %w", existing.Name, err)
+			}
+		} else {
+			// still running
+			return existing.Name, nil
+		}
+	} else if !k8serrors.IsNotFound(err) {
+		return "", fmt.Errorf("get job for node %s: %w", node.Name, err)
+	}
+
+	job := getManagerInstallJobSpecForNode(node, in, operatorImage, licenseSecret, appVersionLabel)
+	if err := cli.Create(ctx, job); err != nil {
+		return "", err
+	}
+
+	return job.Name, nil
+}
+
+func deleteManagerInstallJobs(ctx context.Context, cli client.Client, nodes []corev1.Node) error {
+	for _, node := range nodes {
+		jobName := getManagerInstallJobName(node)
+		job := &batchv1.Job{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: jobNamespace, Name: jobName,
+			},
+		}
+		err := cli.Delete(ctx, job, client.PropagationPolicy(metav1.DeletePropagationBackground))
+		if err != nil {
+			return fmt.Errorf("delete job for node %s: %w", node.Name, err)
+		}
+	}
+
+	return nil
+}
+
+func waitForManagerInstallJobs(ctx context.Context, cli client.Client, nodes []corev1.Node) error {
+	eg := errgroup.Group{}
+
+	for _, node := range nodes {
+		jobName := getManagerInstallJobName(node)
+		eg.Go(func() error {
+			err := waitForManagerInstallJob(ctx, cli, jobName)
+			if err != nil {
+				return fmt.Errorf("wait for job for node %s: %v", node.Name, err)
+			}
+			return nil
+		})
+	}
+
+	// wait cancels
+	err := eg.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func waitForManagerInstallJob(ctx context.Context, cli client.Client, jobName string) error {
+	// 120 steps at 5 second intervals = ~ 10 minutes
```

Comment:
why would it take that long? is it not just downloading a binary to the host and running it?

Reply:
changed to 5 minutes